dev/benchmarks/macrobenchmarks/test_driver/stack_size_perf_test.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' show JsonEncoder;
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:file/file.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart' hide TypeMatcher, isInstanceOf;
+
+import 'package:macrobenchmarks/common.dart';
+
+import 'util.dart';
+
+const JsonEncoder _prettyEncoder = JsonEncoder.withIndent('  ');
+
+void main() {
+  test('stack_size', () async {
+    int stackSizeInByte;
+    await runDriverTestForRoute(kStackSizeRouteName, (FlutterDriver driver) async {
+      final String stackSize = await driver.getText(find.byValueKey(kStackSizeKey));
+      expect(stackSize.isNotEmpty, isTrue);
+      stackSizeInByte= int.parse(stackSize);
+    });
+
+    expect(stackSizeInByte > 0, isTrue);
+
+    await fs.directory(testOutputsDirectory).create(recursive: true);
+    final File file = fs.file(path.join(testOutputsDirectory, 'stack_size.json'));
+    await file.writeAsString(_encodeJson(<String, dynamic>{
+      'stack_size': stackSizeInByte
+    }));
+  }, timeout: const Timeout(kTimeout));
```

Comment:
I thought @Hixie went through and removed all timeouts from our tests...

Reply:
(they're just flakes waiting to happen)

Reply:
This is the general time out for entire benchmarktest. If we don't have this, the test will just hang if something goes wrong. This is not a wait timeout during the test steps, so I think this is not in the scope the timeout @Hixie wanted to remove?

Reply:
Isn't there a timeout enforced by the test runner itself or the agent?

Reply:
yes i think so. I just realized the default timeout is 30s if nothing gets passed in. We can remove this timeout, but it behaves the same unless we remove the change the behavior of `test` function when there is no timeout passes in.